File: src/test/java/cz/vhromada/catalog/web/mapper/MusicMapperTest.java
package cz.vhromada.catalog.web.mapper;

import static org.assertj.core.api.Assertions.assertThat;

import cz.vhromada.catalog.entity.Music;
import cz.vhromada.catalog.web.common.MusicUtils;
import cz.vhromada.catalog.web.fo.MusicFO;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.mapstruct.factory.Mappers;

/**
 * A class represents test for mapper between {@link Music} and {@link MusicFO}.
 *
 * @author Vladimir Hromada
 */
class MusicMapperTest {

    /**
     * Mapper for music
     */
    private MusicMapper mapper;

    /**
     * Initializes mapper.
     */
    @BeforeEach
    void setUp() {
        mapper = Mappers.getMapper(MusicMapper.class);
    }

    /**
     * Test method for {@link MusicMapper#map(Music)}.
     */
    @Test
    void map() {
        final Music music = MusicUtils.getMusic();

        final MusicFO musicFO = mapper.map(music);

        MusicUtils.assertMusicDeepEquals(musicFO, music);
    }

    /**
     * Test method for {@link MusicMapper#map(Music)} with null music.
     */
    @Test
    void map_NullMusic() {
        assertThat(mapper.map(null)).isNull();
    }

    /**
     * Test method for {@link MusicMapper#mapBack(MusicFO)}.
     */
    @Test
    void mapBack() {
        final MusicFO musicFO = MusicUtils.getMusicFO();

        final Music music = mapper.mapBack(musicFO);

        MusicUtils.assertMusicDeepEquals(musicFO, music);
    }

    /**
     * Test method for {@link MusicMapper#mapBack(MusicFO)} with null music.
     */
    @Test
    void mapBack_NullMusic() {
        assertThat(mapper.mapBack(null)).isNull();
    }

}

File: src/test/kotlin/cz/vhromada/catalog/web/mapper/MusicMapperTest.kt
package cz.vhromada.catalog.web.mapper

import cz.vhromada.catalog.entity.Music
import cz.vhromada.catalog.web.CatalogMapperTestConfiguration
import cz.vhromada.catalog.web.common.MusicUtils
import cz.vhromada.catalog.web.fo.MusicFO
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.test.context.ContextConfiguration
import org.springframework.test.context.junit.jupiter.SpringExtension

/**
 * A class represents test for mapper between [Music] and [MusicFO].
 *
 * @author Vladimir Hromada
 */
@ExtendWith(SpringExtension::class)
@ContextConfiguration(classes = [CatalogMapperTestConfiguration::class])
class MusicMapperTest {

    /**
     * Mapper for music
     */
    @Autowired
    private lateinit var mapper: MusicMapper

    /**
     * Test method for [MusicMapper.map].
     */
    @Test
    fun map() {
        val music = MusicUtils.getMusic()

        val musicFO = mapper.map(music)

        MusicUtils.assertMusicDeepEquals(musicFO, music)
    }

    /**
     * Test method for [MusicMapper.mapBack].
     */
    @Test
    fun mapBack() {
        val musicFO = MusicUtils.getMusicFO()

        val music = mapper.mapBack(musicFO)

        MusicUtils.assertMusicDeepEquals(musicFO, music)
    }

}
